### DISCUSSION TITLE
Chrome extension for poster generation

### DISCUSSION THREAD

**Discussion by lucadalli** [google/model-viewer#2761] [Ideas]
The editor is in great shape. Perhaps we could port some of its functionality to a Chrome extension to facilitate poster generation for scenes which use `createTexture`.

This is already quite trivial to do with the `toBlob` method but a Chrome extension could enable operations like bulk poster generation for active model-viewer elements etc.

**Comment by elalish** [comment]
That sounds pretty awesome! I've never made a Chrome extension; are you interested in giving it a try?

**Reply by lucadalli** [reply]
My hands are full at the moment but if you or someone else is keen on taking lead on this, I'd be of assistance wherever I can. I've never worked with Lit before so scaffolding a new project while also juggling my current workload would be arduous, to say the least.

My high-level idea for the extension is a DevTools panel split into two panes:
the left side lists model-viewer elements active on the current page,
the right side shows the operations possible on the selected model-viewer elements.

It's possible to select a single model-viewer element either from the list or by using an element inspect tool similar to the one offered natively by the DevTools.
Selecting a model-viewer element populates the pane on the right side with utilities relevant to a singular model-viewer element (scene graph and material inspection etc).

It's also possible to select multiple model-viewer elements from the list or the element inspect tool by using modifier keys.
Selecting multiple elements populates the right pane with bulk operations possible on those elements, the most obvious being bulk poster generation.

Bulk poster generation accepts two input parameters: poster height and "filename attribute".
The "filename attribute" sets which attribute should be used for the filenames of the poster images (default is "data-poster-name"). 

`<model-viewer data-poster-name="astronaut"  src="shared-assets/models/Astronaut.glb" ... `
would produce poster `astronaut.webp`.
